### PR TITLE
PUD-1545: Calculate partBDueDate off the RTC date, and add 14 days (as today is inclusive in the 15)

### PR DESCRIPTION
**Pull request by philsegal**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallService.kt [hunk 1]
```diff
@@ -96,46 +96,43 @@ class RecallService(
...
+      previousConvictionMainNameCategory = updateRecallRequest.previousConvictionMainNameCategory ?: previousConvictionMainNameCategory,
+      probationInfo = updateRecallRequest.toProbationInfo(this),
+      reasonsForRecall = updateRecallRequest.reasonsForRecall ?: reasonsForRecall,
+      reasonsForRecallOtherDetail = updateRecallRequest.reasonsForRecallOtherDetail ?: reasonsForRecallOtherDetail,
+      recallEmailReceivedDateTime = updateRecallRequest.recallEmailReceivedDateTime ?: recallEmailReceivedDateTime,
+      recallLength = sentencingInfo?.calculateRecallLength(recommendedRecallType),
+      recallNotificationEmailSentDateTime = updateRecallRequest.recallNotificationEmailSentDateTime ?: recallNotificationEmailSentDateTime,
+      sentencingInfo = sentencingInfo,
+      vulnerabilityDiversity = updateRecallRequest.vulnerabilityDiversity ?: vulnerabilityDiversity,
+      vulnerabilityDiversityDetail = updateRecallRequest.vulnerabilityDiversityDetail ?: vulnerabilityDiversityDetail,
```
Comment (philsegal) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallService.kt:135]: Alphabetised

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallServiceTest.kt [hunk 2]
```diff
@@ -58,7 +58,7 @@ class RecallServiceTest {
   private val prisonApiClient = mockk<PrisonApiClient>()
   private val meterRegistry = mockk<MeterRegistry>(relaxed = true)
   private val autoReturnedToCustodyCounter = mockk<Counter>()
-  private val fixedClock = Clock.fixed(Instant.parse("2021-10-04T13:15:50.00Z"), ZoneId.of("UTC"))
+  private val fixedClock = Clock.fixed(Instant.parse("2021-10-04T13:14:50.00Z"), ZoneId.of("UTC"))
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallServiceTest.kt:61]: Unintended change?

**Review by AndyNewton — COMMENTED** [review]
Harmless, but includes one unintended change?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallServiceTest.kt (hunk 2 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallServiceTest.kt:61]: Haha, yes, but makes no difference.

**Review by AndyNewton — APPROVED** [review]
🤷‍♂️

**Pull request merged**